server/src/main/java/org/apache/iotdb/db/mpp/buffer/DataBlockManager.java

```diff
@@ -290,7 +312,7 @@ public ISinkHandle createSinkHandle(
             executorService,
             clientFactory.getDataBlockServiceClient(remoteEndpoint),
             tsBlockSerdeFactory.get(),
-            new SinkHandleListenerImpl(instanceContext));
+            new SinkHandleListenerImpl(instanceContext, instanceContext::failed));
```

Comment:
Now that we use `failureCallback`, we can decouple `FragmentInstanceContext` with this method. I mean, we can change the parameter from `FragmentInstanceContext` to `FailureCallback`

Reply:
You are right.

But to achieve that, we have to introduce a few more callbacks like onFinishedCallback and onClosedCallback. Do you think it is necessary to do it or leaving a TODO is good for now?

Reply:
Sure we can leave a TODO now